beets/autotag/hooks.py

```diff
@@ -67,63 +130,44 @@ class AlbumInfo(object):
     The fields up through ``tracks`` are required. The others are
     optional and may be None.
     """
-    def __init__(self, album, album_id, artist, artist_id, tracks, asin=None,
-                 albumtype=None, va=False, year=None, month=None, day=None,
-                 label=None, mediums=None, artist_sort=None,
-                 releasegroup_id=None, catalognum=None, script=None,
-                 language=None, country=None, albumstatus=None, media=None,
-                 albumdisambig=None, artist_credit=None, original_year=None,
-                 original_month=None, original_day=None, data_source=None,
-                 data_url=None):
-        self.album = album
-        self.album_id = album_id
-        self.artist = artist
-        self.artist_id = artist_id
-        self.tracks = tracks
-        self.asin = asin
-        self.albumtype = albumtype
-        self.va = va
-        self.year = year
-        self.month = month
-        self.day = day
-        self.label = label
-        self.mediums = mediums
-        self.artist_sort = artist_sort
-        self.releasegroup_id = releasegroup_id
-        self.catalognum = catalognum
-        self.script = script
-        self.language = language
-        self.country = country
-        self.albumstatus = albumstatus
-        self.media = media
-        self.albumdisambig = albumdisambig
-        self.artist_credit = artist_credit
-        self.original_year = original_year
-        self.original_month = original_month
-        self.original_day = original_day
-        self.data_source = data_source
-        self.data_url = data_url
-
-    # Work around a bug in python-musicbrainz-ngs that causes some
-    # strings to be bytes rather than Unicode.
-    # https://github.com/alastair/python-musicbrainz-ngs/issues/85
-    def decode(self, codec='utf-8'):
-        """Ensure that all string attributes on this object, and the
+
+    REQ_ATTR = set(('album', 'album_id', 'artist', 'artist_id', 'tracks'))
+
+    def __init__(self, **kwargs):
+        """Sets vars and verifies that REQ_ATTR are in kwargs"""
+        if not self.REQ_ATTR <= set(kwargs.keys()):
+            raise TypeError(
+                "AlbumInfo takes at least those 5 keyword arguments "
+                "{}".format(self.REQ_ATTR))
+        else:
+            super(AlbumInfo, self).__init__(**kwargs)
+
+    def __hash__(self):
+        """Always a good idea to have a hash"""
+        return self['album_id']
+
+    def __reduce__(self):
+        """Used by :py:func:`copy.deepcopy`"""
+        return self['album_id']
```

Comment:
I looked into copy.deepcopy ... you can see it on the pickle doc too:
[https://docs.python.org/2/library/pickle.html#object.__reduce__](url)